dataplane/src/main.rs

```diff
@@ -59,11 +59,12 @@ fn main() {
     /* parse cmd line args */
     let args = CmdArgs::parse();
 
-    let grpc_address = "[::1]:50051".parse().expect("Bad grpc address");
-
-    if let Err(e) = start_mgmt(grpc_address) {
+    // Get the gRPC address from command line args
+    let grpc_addr = args.get_grpc_address();
+    info!("Starting gRPC server on address: {}", grpc_addr);
+    if let Err(e) = start_mgmt(grpc_addr) {
         error!("Failed to start management service: {e}");
-        std::process::exit(0);
+        std::process::exit(0); // TODO - should we exit with an error code?
```

Comment:
personally I would call `panic!("Failed to start management service: {e}")` here

Reply:
added `panic!`